core/src/test/java/kafka/testkit/BrokerNode.java

```diff
@@ -130,7 +154,7 @@ public BrokerNode build(
         this.incarnationId = incarnationId;
         this.initialMetaPropertiesEnsemble = initialMetaPropertiesEnsemble;
         this.combined = combined;
-        this.propertyOverrides = new HashMap<>(propertyOverrides);
+        this.propertyOverrides = Collections.unmodifiableMap(propertyOverrides);
```

Comment:
`ClusterConfig` address the "immutable" by creating immutable objects in Builder. Could we align that behavior?